### DISCUSSION TITLE
How to protect API with OAUTH2 / JWT / Azure B2C authentication?

### DISCUSSION THREAD

**Discussion by biapar** [keystonejs/keystone#6576] [Event Q&A Q3 2021]
How to protect API with OAUTH2 / JWT / Azure B2C authentication?
Thanks.

**Comment by borisno2** [comment]
Nothing out of the box at the moment, you can use my Keystone Next Auth plug-in which adds OAuth to Keystone 6 using Nextauthjs. It has also been tested with Azure B2C - https://github.com/OpenSaasAU/keystone-nextjs-auth. Feel free to reach out if you have any questions.

**Comment by raveling** [comment]
@biapar thanks for your question.
We offered an answer to your question in the recent community Q&A:
https://www.youtube.com/watch?v=r1IJh-iHm1c&t=1462s

**Reply by tv42** [reply]
"This video isn't available anymore"

**Reply by raveling** [reply]
@tv42 so sorry for the confusion – we had an unfortunate situation with our YouTube account and lost that video from the web 🤦 

We intend to fix this and update the timestamp links.

**Reply by biapar** [reply]
Any news?

**Reply by biapar** [reply]
Is there an update?

**Reply by SagnikPradhan** [reply]
Dang, I am interested in this too

**Comment by biapar** [comment]
Any updates?

**Reply by dcousens** [reply]
We don't intend to provide this as an out-of-the-box feature, but if you had any comments to add to the https://github.com/keystonejs/keystone/tree/main/examples/custom-session-jwt example, please let us know
